lib/mongo/operation/create/op_msg.rb

```diff
@@ -31,7 +31,19 @@ class OpMsg < OpMsgBase
 
         def selector(connection)
           # Collation is always supported on 3.6+ servers that would use OP_MSG.
-          spec[:selector].merge(collation: spec[:collation]).compact
+          spec[:selector].merge(
+            collation: spec[:collation],
+            encryptedFields: spec[:encrypted_fields],
+          ).compact.tap do |sel|
+            if sel[:encryptedFields] && sel[:encryptedFields].key?(:fields)
+              sel[:encryptedFields][:fields] = sel[:encryptedFields][:fields].map do |field|
+                if field[:queries] && field[:queries].key?(:contention)
+                  field[:queries][:contention] = BSON::Int64.new(field[:queries][:contention])
```

Comment:
This ultimately mutates user input, doesn't it?

Reply:
If `spec` is a BSON::Document then perhaps it doesn't, but I think there should be a test for this.

Reply:
Added the test